superdesk/io/feed_parsers/nitf.py

```diff
@@ -240,5 +388,42 @@ def get_byline(self, tree):
                 byline = "{} {}".format(byline.strip(), person.text.strip())
         return byline
 
+    def anpa_format_filter(self, elem):
+        anpa_format = elem.get('content').lower() if elem.get('content') is not None else 'x'
+        return CONTENT_TYPE.TEXT if anpa_format == 'x' else CONTENT_TYPE.PREFORMATTED
+
+    def get_original_creator(self, tree):
+        elem = tree.find("head/meta/[@name='aap-original-creator']")
+        if elem is not None:
+            query = {'username': re.compile('^{}$'.format(elem.get('content')), re.IGNORECASE)}
+            user = superdesk.get_resource_service('users').find_one(req=None, **query)
+            if user is not None:
+                return user.get('_id')
+        raise SkipValue()
+
+    def get_version_creator(self, tree):
+        elem = tree.find("head/meta/[@name='aap-version-creator']")
+        if elem is not None:
+            query = {'username': re.compile('^{}$'.format(elem.get('content')), re.IGNORECASE)}
+            user = superdesk.get_resource_service('users').find_one(req=None, **query)
+            if user:
+                return user.get('_id')
+        raise SkipValue()
+
+    def get_task(self, tree):
+        desk_name = tree.find('head/meta[@name="aap-desk"]')
```

Comment:
yes it was hardcoded before the refactoring, I have let it there for now to avoid breaking the current implementation. Actually the idea of my refactoring is to avoid this kind of hardcoding in the future